src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx

```diff
@@ -0,0 +1,174 @@
+  return Promise.all(
+    txs.map(async (tx) => {
+      return await getTransactionDetails(chainId, tx.transaction.id)
+    }),
+  )
+}
+
+const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubmit: (data: null) => void }) => {
+  const [multiSendContract, setMultiSendContract] = useState<Multi_send_call_only>()
+  const [multiSendTxData, setMultiSendTxData] = useState<string>()
+  const [submitError, setSubmitError] = useState<Error | undefined>()
+  const chain = useCurrentChain()
+  const { safe } = useSafeInfo()
+  const provider = useWeb3()
+
+  const [txsWithDetails, error, loading] = useAsync<TransactionDetails[]>(() => {
+    if (!chain?.chainId) return
+
+    return getTxsWithDetails(data.txs, chain.chainId)
+  }, [data.txs, chain?.chainId])
+
+  useEffect(() => {
+    if (!chain) return
+
+    setMultiSendContract(getMultiSendCallOnlyContractInstance(chain.chainId))
+  }, [chain])
+
+  useEffect(() => {
+    if (!txsWithDetails || !chain) return
+
+    const multiSendTxs = getMultiSendTxs(txsWithDetails, chain, safe.address.value, safe.version)
+    const data = encodeMultiSendData(multiSendTxs)
+
+    setMultiSendTxData(data)
+  }, [chain, safe.address.value, safe.version, txsWithDetails])
+
+  const sendTx = async () => {
+    setSubmitError(undefined)
+    if (!provider || !multiSendTxData || !chain || !multiSendContract || !txsWithDetails) return
+
+    try {
+      const multiSendTransaction = await multiSendContract.connect(provider.getSigner()).multiSend(multiSendTxData)
+
+      txsWithDetails.forEach((tx) => {
+        createExistingTx(chain.chainId, safe.address.value, tx.txId, tx).then((safeTx) =>
+          txDispatch(TxEvent.MINING, { txId: tx.txId, txHash: multiSendTransaction.hash, tx: safeTx }),
+        )
+      })
+
+      onSubmit(null)
+    } catch (err) {
+      logError(Errors._804, (err as Error).message)
+      setSubmitError(err as Error)
+    }
+  }
+
+  return (
+    <div>
+      <DialogContent>
+        <Typography variant="body2" mb={2}>
+          This transaction batches a total of {data.txs.length} transactions from your queue into a single Ethereum
+          transaction. Please check every included transaction carefully, especially if you have rejection transactions,
+          and make sure you want to execute all of them. Included transactions are highlighted in green when you hover
+          over the execute button.
+        </Typography>
+        <Typography color="secondary.light">Interact with:</Typography>
+        {multiSendContract && (
+          <EthHashInfo address={multiSendContract.address} shortAddress={false} hasExplorer showCopyButton />
+        )}
+        {multiSendTxData && (
+          <>
+            <Typography mt={2} color="secondary.light">
+              Data (hex encoded)
+            </Typography>
+            {generateDataRowValue(multiSendTxData, 'rawData')}
+          </>
+        )}
```

Comment:
What do you think about including skeletons here?

Reply:
I changed it from a useEffect to useMemo as @katspaugh suggested which makes it available on the first render without a loading state.